mbed-drivers/I2C.h

```diff
@@ -48,7 +49,7 @@ namespace mbed {
  * }
  * @endcode
  */
-class I2C {
+__deprecated class I2C {
```

Comment:
This I2C class shall be deprecated for v1, but not for v0, is that correct? 

Is deprecated required for all methods? Not sufficient once you defined I2C object to get deprecated warning?